### PR TITLE
Fix text/plain + BOM in decoder + Response_encoding parameter

### PR DESCRIPTION
**Pull request by lmazuel**
Fix https://github.com/Azure/azure-sdk-for-python/issues/9589

### CODE REVIEW

File: sdk/core/azure-core/azure/core/pipeline/policies/_universal.py [hunk 1]
```diff
@@ -538,7 +540,8 @@ def deserialize_from_http_generics(
         else:
             mime_type = "application/json"
 
-        return cls.deserialize_from_text(response.text(), mime_type, response=response)
+        # FIXME should extract charset from content-type
+        return cls.deserialize_from_text(response.text(encoding="utf-8-sig"), mime_type, response=response)
```
Comment (lmazuel) [inline sdk/core/azure-core/azure/core/pipeline/policies/_universal.py]: Should fix the FIXME actually

**Comment by lmazuel** [comment]
Will need https://github.com/Azure/azure-sdk-for-python/pull/9598 to get CI green

**Comment by lmazuel** [comment]
Related to https://github.com/Azure/azure-sdk-for-python/pull/9203

**Comment by lmazuel** [comment]
Will need https://github.com/Azure/azure-sdk-for-python/pull/9606 to get CI green

**Comment by lmazuel** [comment]
Interesting discussion:
https://github.com/psf/requests/issues/2086

**Comment by bryevdv** [comment]
Gotta love multi-many-year issues. :) 

LGTM AFAICT

File: sdk/core/azure-core/azure/core/pipeline/policies/_universal.py [hunk 2]
```diff
@@ -538,7 +549,15 @@ def deserialize_from_http_generics(
...
-        return cls.deserialize_from_text(response.text(), mime_type, response=response)
+        # Rely on transport implementation to give me "text()" decoded correctly
+        return cls.deserialize_from_text(response.text(encoding), mime_type, response=response)
+
+    def on_request(self, request):
+        # type: (PipelineRequest) -> None
+        options = request.context.options
+        response_encoding = options.pop("response_encoding", self._response_encoding)
+        if response_encoding:
+            request.context.setdefault("response_encoding", response_encoding)
```
Comment (johanste) [inline sdk/core/azure-core/azure/core/pipeline/policies/_universal.py]: Under what circumstance would there already be a `response_encoding` provided in the `request.context`? And wouldn't we want to replace it if the user explicitly passed on in?

Comment (lmazuel) [reply sdk/core/azure-core/azure/core/pipeline/policies/_universal.py]: I don't see any scenario where we pop something AND it's also in the context at the same time. Context will be filled with previous pop if we pass twice in the on_request (e.g. this policy deeper than retry).
I'm fine to force it if we pop something, sounds fair behavior to me.

**Pull request merged**